src/main/java/edu/kis/powp/jobs2d/command/CommandCounterVisitor.java

```diff
@@ -0,0 +1,35 @@
+package edu.kis.powp.jobs2d.command;
+
+public class CommandCounterVisitor implements CommandVisitor {
+	private int operateToCount = 0;
+
+	private int setPositionCount = 0;
+	private int iCompoundCommand = 0;
+
+	public int getOperateToCount() {
+		return operateToCount;
+	}
+
+	public int getSetPositionCount() {
+		return setPositionCount;
+	}
+
+	public int getICompoundCommand() {
+		return iCompoundCommand;
+	}
+
+	@Override
+	public void visit(OperateToCommand operateToCommand) {
+		operateToCount++;
+	}
+
+	@Override
+	public void visit(SetPositionCommand setPositionCommand) {
+		setPositionCount++;
+	}
```

Comment:
This is not a stateless approach, which is acceptable, but not preferred.
Better solution would set operateToCount = 0 or 1 and setPositionCount = 0 or 1, because e.g., the actual number of commands in a single OperateToCommand is one operateTo and none setPosition. The trouble is, that a more complex logic should be implemented in public void visit(ICompoundCommand compoundCommand), because it needs to iterate and aggregate the results of visitation.